lesson25/main.js

```diff
@@ -19,14 +32,84 @@ submitBtn.addEventListener(
   false
 );
 
+const checkInputVal = (targetForm) => {
+  const targetVal = targetForm.value;
+  let result;
+
+  switch (targetForm.id) {
+    case "userName":
+      const minLength = 1;
+      const maxLength = 15;
+      result = targetVal.length >= minLength && targetVal.length <= maxLength;
+      break;
+    case "email":
+      const emailValid = /^\w[\w_.-]*@.+\..+/;
+      result = emailValid.test(targetVal);
+      break;
+    case "pass":
+      const passValid = /(?=.*[a-z])(?=.*[A-Z])(?=.*\d)\w{8,}/;
+      result = passValid.test(targetVal);
+      break;
+    default:
+      console.error(`は見つかりませんでした`);
+  }
+
+  result ? (flags[targetForm.id] = true) : (flags[targetForm.id] = false);
+  return result;
+};
```

Comment:
IMO
The process is similar to createErrorMessage()
I feel like there is redundancy.

The Validation process in one place.
```js
const validations = {
  userName: {
    maxLength: 15,
    minLength: 3,
    isValid: (value) => {
      return (
        value.length >= validations.userName.minLength &&
        value.length <= validations.userName.maxLength
      );
    }
  },
  email: {
    errorMessage: "メールアドレスが空欄・もしくは形式が異なっています。",
    isValid: (value) => /.+@.+\..+/.test(value)
  },
  pass: {
    errorMessage: "8文字以上の大小の英数字を交ぜたものにしてください。",
    isValid: (value) =>
      /^(?=.*?[a-z])(?=.*?[A-Z])(?=.*?\d)[a-zA-Z\d]{8,}$/.test(value)
  }
};
```
Based on that, I also made some edits to the following functions
```js
const checkInputVal = (targetForm) => {
  const targetInput = document.getElementById(targetForm.id);
  const result = validations[targetInput].isValid(targetForm.value);
  result ? (flags[targetInput] = true) : createErrorMessage(targetInput);
};

const createErrorMessage = (targetInput) => {
  const errorTxt = createElementWithClassName("p", "error-txt");
  errorTxt.textContent = validations[targetInput].errorMessage;
  targetInput.after(errorTxt);
};

//It's called in the HTML file
const validateInputVal = (targetForm) => {
  const errorTxtEle = targetForm.nextElementSibling;
  checkInputVal(targetForm);
  errorTxtEle && errorTxtEle.remove();
  switchSbumitBtn();
};
```
[codesandbox](https://codesandbox.io/s/lesson25-forked-qs003?file=/main.js)
※The implementation part of Validation was taught to me by Mr. Morita.

Reply:
It looks much better! Simple, not complicated 👏 
Regular expressions part is also simpler, especially e-mail one.
Thank you so much for teaching me!
[2761d46](https://github.com/yuka830/js-lessons/pull/30/commits/2761d462d31116b55a9625f5f70e7db8a411f962)